mantidproject/mantid
# Fix Issues with SANS Systemtest on Conda Builds

**Description of work.**
The reduction when using monitor 4 in a SANS2D reduction was producing a massively negative spike in the output workspace. This output is meant to be a cross-section so is not meant to contain any negative values. Why this was happening is unclear. NaNs and inf values are already filtered out of the output when they occur, by being set to 0.0 as a default. In order to get the tests to function properly, negative values in the output will also be set to 0. Hopefully replacing the data files with some new ones will fix the issue, but the filtering is still useful. 

This also adds some functionality to the `ReplaceSpecialValues` algorithm to allow the big and small values to be adjusted without being converted to absolute first. 

**To test:**
Sys test:
1. Run the `SANS2D_GDW20_4m_cycle_22_02` system test and check it passes. Ideally this needs to be checked on conda and non-conda builds on each system (windows especially, as the error was greater there). 

Algorithm:
1. Load a workspace or make a sample one with some positive and negative values
2. Check that when UseAbsolute is ticked (which it should be by default) that negative values are adjusted when below the SmallNumberThreshold 

*There is no associated issue.*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
